### PR TITLE
Categories form

### PR DESCRIPTION
**Pull request by MarlenAw**
issue #2

### CODE REVIEW

**Review by MynahMarie — APPROVED** [review]
I did the work with @MarlenAw so I have nothing to comment on this, looks good to me!

**Review by hoslack — APPROVED** [review]

File: client/src/index.js [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
 ReactDOM.render(
-<Provider store={store}>
-  <App />
-</Provider>,
-  document.getElementById('root'));
+  <Provider store={store}>
+    <App />
+  </Provider>,
+  document.getElementById("root")
+);
```
Comment (Karyum) [inline client/src/index.js]: make sure you have the same prettier settings!!
in this case single qoutes for strings

File: client/package.json [hunk 2]
```diff
@@ -9,6 +9,7 @@
     "react-router-dom": "^4.2.2",
     "react-scripts": "1.1.0",
     "redux": "^3.7.2",
+    "redux-promise": "^0.5.3",
```
Comment (Karyum) [inline client/package.json]: redux-promise is not being used here, even though you wanted to use it in the future you would atleast need to have `redux-thunk` :sweat_smile:

File: client/src/containers/CategoryForm.js [hunk 3]
```diff
@@ -1,21 +1,43 @@
 import React, { Component } from "react";
 import { connect } from "react-redux";
+import * as actions from "../actions";
```
Comment (Karyum) [inline client/src/containers/CategoryForm.js:3]: i would feel this my self preference but i prefer to do destructing here to get the specific action that i want, but you don't have to change this i think its more self preference thing

File: client/src/reducers/index.js [hunk 4]
```diff
@@ -1,6 +1,8 @@
-import { combineReducers } from 'redux';
-import categoryList from './category_list';
+import { combineReducers } from "redux";
+import categoryList from "./category_list";
+import categoryReducer from "./categoryReducer";
 
 export default combineReducers({
-  categories: categoryList
+  categories: categoryList,
+  category: categoryReducer
```
Comment (Karyum) [inline client/src/reducers/index.js:7]: so correct me if im wrong but this part of the will just hold a single category correct?

File: client/src/containers/CategoryForm.js [hunk 5]
```diff
@@ -33,6 +55,13 @@ class CategoryForm extends Component {
   }
 }
 
-const mapStateToProps = ({ categories }) => ({ categories });
+const mapStateToProps = ({ categories, category }) => ({
+  categories,
+  category
```
Comment (Karyum) [inline client/src/containers/CategoryForm.js]: are you guys using this somewhere in the component which i can't find ? :sweat_smile:

**Review by Karyum — CHANGES_REQUESTED** [review]
# Redux mindfuck FTW

File: client/src/containers/CategoryForm.js (hunk 5 above)
Comment (MarlenAw) [reply client/src/containers/CategoryForm.js]: Lol no we're not using it 😅

File: client/src/reducers/index.js (hunk 4 above)
Comment (MarlenAw) [reply client/src/reducers/index.js:7]: On line 7, yes

**Review by Karyum — APPROVED** [review]

**Pull request merged**